lib/jwt.service.ts

```diff
@@ -22,18 +22,24 @@ export class JwtService {
   verify<T extends object = any>(
     token: string,
     options?: jwt.VerifyOptions
-  ): T {
+  ): void {
     const verifyOptions = options
       ? {
           ...(this.options.verifyOptions || {}),
           ...options
         }
       : this.options.verifyOptions;
-    return jwt.verify(
+     return jwt.verify(
       token,
       this.options.publicKey || (this.options.secretOrPrivateKey as any),
-      verifyOptions
-    ) as T;
+      verifyOptions,
+      (error, jwtPayload) => {
+        if (error) {
+          throw error;
+        }
+        return jwtPayload;
```

Comment:
This will not return the `jwtPayload` from the outer function. You would either have to have another callback for the function calling `JwtService.verify()` or return a promise instead.

This should also be considered a breaking change as it changes the behavior of a public function.